runtime/vm/classsupport.c

```diff
@@ -1092,10 +1092,30 @@ loadWarmClassFromSnapshot(J9VMThread *vmThread, J9ClassLoader *classLoader, J9Cl
 	rc = TRUE;
 
 done:
+
 	return rc;
 }
 #endif /* defined(J9VM_OPT_SNAPSHOTS) */
 
+J9Class *
+segmentIteratorNextClass(J9ClassSegmentWalkState *state)
+{
+	for (;;) {
+		J9Class *current = state->startClass;
+
+		if (NULL == current) {
+			return NULL;
+		}
```

Comment:
I'm not sure all compilers will recognize that this always returns a value; suggest `break;` here, `return NULL;` after the loop.